File: cityfinder.js
async function getCityFromLatLng(latitude, longitude) {
    const apiKey = 'Akg2tJWLm6UJu1jjM-aF_OGgQDkJC3xbDV6xgimSckEaBw-ShyHZrWE1MQm0J6o8'; // Replace with your Bing Maps API key
    const apiUrl = `https://dev.virtualearth.net/REST/v1/Locations/${latitude},${longitude}?o=json&key=${apiKey}`;
  
    try {
      const response = await fetch(apiUrl);
      const data = await response.json();
  
      if (data.resourceSets && data.resourceSets.length > 0) {
        const location = data.resourceSets[0].resources[0];
        if (location && location.address && location.address.locality) {
          return location.address.locality; // Return the city
        }
      }
  
      throw new Error('Could not determine city.');
    } catch (error) {
      console.error('Error fetching data:', error);
      throw new Error('Error fetching city data.');
    }
  }
  
//   // Example usage
//   const latitude = 49.269802;
//   const longitude = -123.083763;
  
//   getCityFromLatLng(latitude, longitude)
//     .then(city => console.log('City:', city))
//     .catch(error => console.error('Error:', error));
  

File: server/cityfinder.py
import requests

def get_city_from_lat_lng(latitude, longitude):
    # Bing Maps API endpoint for reverse geocoding
    api_key = "Akg2tJWLm6UJu1jjM-aF_OGgQDkJC3xbDV6xgimSckEaBw-ShyHZrWE1MQm0J6o8"
    url = f'https://dev.virtualearth.net/REST/v1/Locations/{latitude},{longitude}?o=json&key={api_key}'

    try:
        # Send a GET request to the Bing Maps API
        response = requests.get(url)
        data = response.json()

        if response.status_code == 200 and 'resourceSets' in data and data['resourceSets']:
            resources = data['resourceSets'][0].get('resources', [])
            if resources:
                address = resources[0].get('address', {})
                if 'locality' in address:
                    return address['locality']

        raise ValueError('Could not determine city.')

    except requests.RequestException as e:
        raise ValueError(f'Error fetching data: {e}')

# latitude = 37.7749
# longitude = -122.4194

# try:
#     city = get_city_from_lat_lng(latitude, longitude)
#     print('City:', city)
# except ValueError as e:
#     print('Error:', str(e))

